icecream/icecream.py

```diff
@@ -154,15 +155,38 @@ def format_pair(prefix, arg, value):
     return '\n'.join(lines)
 
 
-def argumentToString(obj):
-    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj)
+def argumentToString(obj, width=DEFAULT_LINE_WRAP_WIDTH):
+    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj, width=width)
     s = s.replace('\\n', '\n')  # Preserve string newlines in output.
     return s
 
 
+def detect_terminal_width(prefix, default=DEFAULT_LINE_WRAP_WIDTH):
+    """ Returns the number of columns that this terminal can handle. """
+    width = default
+    try:
+        if hasattr(shutil, "get_terminal_size"):
+            width = shutil.get_terminal_size().columns
+        else:  # Python 2.x doesn't support get_terminal_size
+            from backports.shutil_get_terminal_size import get_terminal_size
+            width = get_terminal_size().columns
+    except OSError:  # Not in TTY
```

Comment:
Looks like the shutil version doesn't raise OSError any more. But maybe the whole thing should be wrapped in `except Exception` just in case something weird happens.

Also you should pass the default width as a fallback to `get_terminal_size`. Otherwise it's going to use its own default fallback (80 columns) when the other methods fail.

Reply:
Will do on both counts.